pygenometracks/tracksClass.py

```diff
@@ -682,23 +698,21 @@ def plot(self, ax, chrom_region, region_start, region_end):
                       for x in ticks]
             labels[-2] += " Mbp"
 
-        if self.properties['where'] == 'top':
-            ax.axis["x"] = ax.new_floating_axis(0, 0.2)
-            ax.axis["x"].set_axis_direction("top")
-            label_y_pos = 0.99
-            vert_align = 'top'
-        else:
-            ax.axis["x"] = ax.new_floating_axis(0, 0.9)
-            label_y_pos = 0.01
-            vert_align = 'bottom'
-        ax.text(0.5, label_y_pos, chrom_region, horizontalalignment='center',
-                fontsize=self.properties['fontsize'],
-                verticalalignment=vert_align, transform=ax.transAxes)
+        # Fix ticks values
+        ax.set_xticks(ticks)
+
+        # Fix the limits which can be overwritten by set_xticks:
+        ax.set_xlim(start, end)
+
+        # Put the label
+        ax.set_xticklabels(labels)
 
-        ax.axis["x"].axis.set_ticklabels(labels)
         ax.axis['x'].axis.set_tick_params(which='minor', bottom='on')
```

Comment:
Not sure if this is our fault but this basically reads 'axis on axis on axis apply xyz'. Maybe that naming scheme can be changed?

Reply:
In fact I don't know what this line is doing because I don't think we have minor ticks on the x-axis...